autogen/agentchat/groupchat.py

```diff
@@ -36,26 +36,21 @@ class GroupChat:
         When set to True and when a message is a function call suggestion,
         the next speaker will be chosen from an agent which contains the corresponding function name
         in its `function_map`.
-    - custom_speaker_selection_func: pass a function to customize the speaker selection. Default is None.
-        custom_speaker_selection_func:
-            Parameters:
-                - last_speaker: Agent
-                    The last speaker in the group chat.
-                - agents: List[Agents]
-                    All agents in the group chat.
-                - messages: List[Dict]
-                    All past messages in the group chat.
-            Returns:
-                - Agent
-                    The selected speaker.
-        The return agent should be one of the agents in the group chat.
-        If the return agent is not valid, we will use default to the method specified in `speaker_selection_method`.
     - speaker_selection_method: the method for selecting the next speaker. Default is "auto".
         Could be any of the following (case insensitive), will raise ValueError if not recognized:
         - "auto": the next speaker is selected automatically by LLM.
         - "manual": the next speaker is selected manually by user input.
         - "random": the next speaker is selected randomly.
         - "round_robin": the next speaker is selected in a round robin fashion, i.e., iterating in the same order as provided in `agents`.
+        - a customized speaker selection function: if passed in a function, it will be called with the following parameters:
+            custom_speaker_selection_func:
+            Parameters:
+                - last_speaker: Agent
+                    The last speaker in the group chat.
+                - messages: List[Dict]
+                    All past messages in the group chat.
```

Comment:
Are these enough? I think we should pass the groupchat object instead of messages.

Reply:
Yes, I checked the variables in GroupChat, and I don't think they are useful for determining the next states. They are mainly parameters for different built-in speaker selection methods. The static transition mostly rely on string matching in messages.

Reply:
what about the agents? Don't you need to know the agents at least?
And in case graphs of speaker order are given, don't you need to know those too?

Reply:
Hey @kevin666aa, similar to sonichi's point, I wasn't able to successfully perform that test in `test_custom_speaker_selection_with_transition_graph` of 251cba6. I think the problem is that custom_speaker_selection_func must return an Agent, but I think perhaps sometimes we need to be able to return None (L1015 of test_groupchat.py)?

Hoping the test case makes the problem easier to visualize. :)

Reply:
Thanks for the updated test. I will change the function to let it pass in a groupchat object.